js/core/utils/data.js

```diff
@@ -191,7 +191,13 @@ export const toComparable = function(value, caseSensitive) {
     }
 
     if(!caseSensitive && typeof value === 'string') {
-        return value.toLowerCase();
+        if(options?.collatorOptions?.sensitivity === 'base') {
+            const REMOVE_DIACRITICAL_MARKS_REGEXP = /[\u0300-\u036f]/g;
+
+            value = value.normalize('NFD').replace(REMOVE_DIACRITICAL_MARKS_REGEXP, '');
+        }
+
+        return options?.locale ? value.toLocaleLowerCase(options.locale) : value.toLowerCase();
```

Comment:
Minor comment

Minor comment

Let's remove the first redundant ```?```
(because options always defined and equals ```{}``` by default)
```suggestion
        return options.locale ? value.toLocaleLowerCase(options.locale) : value.toLowerCase();
```